server/account/account.go

```diff
@@ -52,6 +45,30 @@ func (s *Server) UpdatePassword(ctx context.Context, q *account.UpdatePasswordRe
 		updatedUsername = q.Name
 	}
 
+	if session.Iss(ctx) == session.SessionManagerClaimsIssuer {
+		// local user is changing own password or another user password
+
+		// using is changing own password. // ensure token belongs to a user, not project
+		if username == updatedUsername && rbacpolicy.IsProjectSubject(username) {
+			return nil, status.Errorf(codes.InvalidArgument, "password can only be changed for local users, not user %q", username)
+		}
+
+		err := s.sessionMgr.VerifyUsernamePassword(username, q.CurrentPassword)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "current password does not match")
+		}
+	} else {
+		// SSO user is changing or local user password
```

Comment:
Where do we make sure that SSO user has enough permissions here?

Reply:
This is verified here: https://github.com/argoproj/argo-cd/pull/3298/files#diff-3dacf6e808d43913c38006b06308a2a5R41

If user is not trying to change own password we verify that user has permission to update account <account-name>.

Reply:
Ah, now I see. But I think that check can be circumvented when the SSO username is the same as the local username, right? So, when there's a local user "jannfis", and someone has the SSO account "jannfis", he can change the local user's password without RBAC enforcement. Or is `sub` something different than the plain username in case of SSO?

Reply:
oh, nice catch. We need to check issuers. Working on the fix.

Reply:
implemented the fix. Now we check permissions unless the local user is trying to change own password.